executor/partition_table_test.go

```diff
@@ -176,6 +178,48 @@ PRIMARY KEY (pk1,pk2)) partition by hash(pk2) partitions 4;`)
 	tk.MustQuery("select /*+ INL_MERGE_JOIN(dt, rr) */ * from coverage_dt dt join coverage_rr rr on (dt.pk1 = rr.pk1 and dt.pk2 = rr.pk2);").Sort().Check(testkit.Rows("ios 3 ios 3 2", "linux 5 linux 5 1"))
 }
 
+func (s *partitionTableSuite) TestPartitionInfoDisable(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t_info_null")
+	tk.MustExec(`CREATE TABLE t_info_null (
+  id bigint(20) unsigned NOT NULL AUTO_INCREMENT,
+  date date NOT NULL,
+  media varchar(32) NOT NULL DEFAULT '0',
+  app varchar(32) NOT NULL DEFAULT '',
+  xxx bigint(20) NOT NULL DEFAULT '0',
+  PRIMARY KEY (id, date),
+  UNIQUE KEY idx_media_id (media, date, app)
+) PARTITION BY RANGE COLUMNS(date) (
+  PARTITION p201912 VALUES LESS THAN ("2020-01-01"),
+  PARTITION p202001 VALUES LESS THAN ("2020-02-01"),
+  PARTITION p202002 VALUES LESS THAN ("2020-03-01"),
+  PARTITION p202003 VALUES LESS THAN ("2020-04-01"),
+  PARTITION p202004 VALUES LESS THAN ("2020-05-01"),
+  PARTITION p202005 VALUES LESS THAN ("2020-06-01"),
+  PARTITION p202006 VALUES LESS THAN ("2020-07-01"),
+  PARTITION p202007 VALUES LESS THAN ("2020-08-01"),
+  PARTITION p202008 VALUES LESS THAN ("2020-09-01"),
+  PARTITION p202009 VALUES LESS THAN ("2020-10-01"),
+  PARTITION p202010 VALUES LESS THAN ("2020-11-01"),
+  PARTITION p202011 VALUES LESS THAN ("2020-12-01")
+)`)
+	is := infoschema.GetInfoSchema(tk.Se)
+	tbl, err := is.TableByName(model.NewCIStr("test"), model.NewCIStr("t_info_null"))
+	c.Assert(err, IsNil)
+
+	tbInfo := tbl.Meta()
+	// Mock for a case that the tableInfo.Partition is not nil, but tableInfo.Partition.Enable is false.
+	// That may happen when upgrading from a old version TiDB.
+	tbInfo.Partition.Enable = false
+	tbInfo.Partition.Num = 0
+
+	tk.MustExec("set @@tidb_partition_prune_mode = 'static'")
+	tk.MustQuery("explain select * from t_info_null where (date = '2020-10-02' or date = '2020-10-06') and app = 'xxx' and media = '19003006'").Check(testkit.Rows("Batch_Point_Get_5 2.00 root table:t_info_null, index:idx_media_id(media, date, app) keep order:false, desc:false"))
```

Comment:
And `explain select * from t_info_null`. It shouldn't include `PartitionUnion`.

Reply:
done